python/seldon_core/metrics.py

```diff
@@ -127,24 +127,54 @@ def update(self, custom_metrics: List[Dict], method: str):
 
         for metrics in custom_metrics:
             metrics_type = metrics.get("type", "COUNTER")
+
             tags = metrics.get("tags", {})
+
             tags["method"] = method
-            key = (metrics_type, metrics["key"], SeldonMetrics._generate_tags_key(tags))
-            # Add tag that specifies which method added the metrics
+
+            worker_data_key = (
+                metrics_type,
+                metrics["key"],
+                SeldonMetrics._generate_tags_key(tags),
+            )
+
             if metrics_type == "COUNTER":
-                value = worker_data.get(key, {}).get("value", 0)
-                worker_data[key] = {"value": value + metrics["value"], "tags": tags}
-            elif metrics_type == "TIMER":
-                vals, sumv = worker_data.get(key, {}).get(
-                    "value", (list(np.zeros(len(BINS) - 1)), 0)
+                value = worker_data.get(worker_data_key, {}).get("value", 0)
+                worker_data[worker_data_key] = {
+                    "value": value + metrics["value"],
+                    "tags": tags,
+                }
+            elif metrics_type == "HISTOGRAM":
+                bins = metrics.get("bins")
+
+                vals, sumv = worker_data.get(worker_data_key, {}).get(
+                    "value", (np.zeros(len(bins) - 1).tolist(), 0)
+                )
+
+                bin_vals = metrics["value"]
+                worker_data[worker_data_key] = {
+                    "value": self._update_hist(bin_vals, vals, sumv, bins),
+                    "tags": tags,
+                }
+            elif metrics_type == "TIMER":  # prefer to use HISTOGRAM instead
+                bins = BINS
+
+                vals, sumv = worker_data.get(worker_data_key, {}).get(
+                    "value",
+                    (np.zeros(len(bins) - 1).tolist(), 0),
                 )
+
                 # Dividing by 1000 because unit is milliseconds
-                worker_data[key] = {
-                    "value": self._update_hist(metrics["value"] / 1000, vals, sumv),
+                bin_val = metrics["value"] / 1000
```

Comment:
If the only difference between `TIMER` and new alias `HISTOGRAM` is factor of 1000 they should both fall into 
```
elif metrics_type == "HISTOGRAM" or metrics_type == "TIMER"
```
with one small `if` statement at place where value needs to be modified by factor `1000`.